app/app.rb

```diff
@@ -34,6 +34,18 @@
   body {}
 end
 
+get '/devotionals' do
+  begin
+    devotionals = Devotional.all
+
+    content_type :json
+    devotionals.to_json
+  rescue StandardError => e
```

Comment:
Cool, this is _ok_... But think about what scenarios would result in an error here, and make sure that we have a test for that scenario.

StandardError is the ancestor/superclass for all kinds of errors (check out
https://www.exceptionalcreatures.com/bestiary.html for some more information on these)

Maybe this rescue block should be applied more generally, rather than just in this block.
Check out how we're handling the KeyError at the bottom of this file. 
Catching errors this way will apply it to ANY request that raises it, which may be an improvement to the API overall!

Reply:
Just changed it back to have a similar structure to /home except using the StandardError. Is that what you meant?

Reply:
No, not quite.
> Maybe this rescue block should be applied more generally, rather than just in this block.
> Check out how we're handling the KeyError at the bottom of this file.

 I was suggesting that instead of only rescuing StandardError within the /devotionals route, we use a strategy more like this, which we have at the bottom of this file: https://github.com/ittullos/pray4tn/blob/5f2cd6e59b180a12f6db16c589ed977bb02db917/app/app.rb#L86

> Catching errors this way will apply it to ANY request that raises it, which may be an improvement to the API overall!

Reply:
Ahh okay gotcha. So I should make a similar structure at the bottom for StandardError? And this happens for the entire file, meaning I wouldn't have to call it here in the method? 

Also is that considered a method at the bottom of the file? Never seen the "error" definition before

Reply:
We can jump on a call and talk through this if you want, it might be easier than commenting back and forth.
This `error SomeErrorClass do` thing is from Sinatra: https://github.com/sinatra/sinatra?tab=readme-ov-file#error-handling
It basically works like a rescue block that applies to all the routes. This way you can avoid duplicating every possible error case in every route block, and you can count on a consistent API response anytime that error occurs.